frontend/src/modals/upload-document/UploadDocumentModal.tsx

```diff
@@ -64,38 +76,86 @@ export const UploadDocumentModal = ({ projectId, onDocumentUploaded, onClose })
       },
     };
     try {
-      const response = await sendFile(uploadDocumentMutation, variables, file);
-      const action = { type: HANDLE_SUBMIT_ACTION, response };
-      dispatch(action);
+      const { data, error } = await sendFile(uploadDocumentMutationFile, variables, file);
+      if (error) {
+        const showToastEvent: ShowToastEvent = {
+          type: 'SHOW_TOAST',
+          message: error.message,
+        };
+        dispatch(showToastEvent);
+      } else if (data) {
```

Comment:
No else if, we can have both error and data

Reply:
Done